### PR TITLE
Add Axon Keyboard

### PR DESCRIPTION
**Pull request by thatfellarobin**
## Description

Add Axon, my DIY keyboard.
Reddit post about it: https://www.reddit.com/r/MechanicalKeyboards/comments/n0fhxn/axon_my_diy_40_throughhole_keyboard/

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/axon/config.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
```
Comment (fauxpark) [inline keyboards/handwired/axon/config.h:54]:
```suggestion
```

File: keyboards/handwired/axon/readme.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Axon
+
+![Axon](https://i.imgur.com/VRVv9i9.jpeg)
+*Axon v1.0 built with Durock T1 and GMK Bento*
```
Comment (fauxpark) [inline keyboards/handwired/axon/readme.md:4]:
```suggestion
![Axon](https://i.imgur.com/VRVv9i9h.jpeg)

Axon v1.0 built with Durock T1 and GMK Bento
```

File: keyboards/handwired/axon/readme.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+* Keyboard Maintainer: [Robin Liu](https://github.com/thatfellarobin)
+
+Make example for this keyboard (after setting up your build environment):
+
+`make handwired/axon:default`
+
+alternative:
+
+`qmk compile -kb handwired/axon -km default`
```
Comment (fauxpark) [inline keyboards/handwired/axon/readme.md:16]:
```suggestion
    make handwired/axon:default
```

File: keyboards/handwired/axon/config.h [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
```
Comment (fauxpark) [inline keyboards/handwired/axon/config.h:56]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/axon/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,26 @@
+#include "axon.h"
```
Comment (drashna) [inline keyboards/handwired/axon/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/axon/keymaps/default/keymap.c (hunk 5 above)
Comment (thatfellarobin) [reply keyboards/handwired/axon/keymaps/default/keymap.c:18]: Resolved in commit 7a0f99f: https://github.com/qmk/qmk_firmware/pull/12879/commits/7a0f99fe977601d849683a9f77a6aefd1e8f33bb

File: keyboards/handwired/axon/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "axon.h"
```
Comment (fauxpark) [inline keyboards/handwired/axon/keymaps/default/keymap.c:18]:
```suggestion
#include QMK_KEYBOARD_H
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by thatfellarobin** [comment]
- re-rebased, with credit to fauxpark for suggested changes (thanks!)

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**